builder/builder-next/exporter/wrapper.go

```diff
@@ -82,7 +82,7 @@ func (i *imageExporterInstanceWrapper) Export(ctx context.Context, src *exporter
 	}
 
 	if i.callbacks.Named != nil {
-		for _, name := range strings.Split(out[string(exptypes.OptKeyName)], ",") {
+		for _, name := range strings.Split(out["image.name"], ",") {
```

Comment:
Can you also check that this key exists and is non-empty?

We also recently added a constant to buildkit for this because we noticed this wasn't working properly. The constant was added in https://github.com/moby/buildkit/pull/5862. If it's possible to use that, it would be preferred. Otherwise a note that the exported constant should be used when buildkit is updated.

Reply:
Thanks, done